rest/oidc_api_test.go

```diff
@@ -1285,6 +1285,80 @@ func TestUserAPIReadOnlyFields(t *testing.T) {
 	assert.Equal(t, userOutput, resp.Body.String())
 }
 
+// TestAdminAndJWTChannels validates interaction between admin channels and JWT channels, to ensure each are preserved when
+// the other is changed
+func TestAdminAndJWTChannels(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyAll)
+
+	testProviders := auth.OIDCProviderMap{
+		"foo": mockProviderWith("foo", mockProviderUserPrefix{"foo"}, mockProviderChannelsClaim{"channels"}),
+	}
+	defaultProvider := "foo"
+
+	mockAuthServer, err := newMockAuthServer()
+	require.NoError(t, err, "Error creating mock oauth2 server")
+	mockAuthServer.Start()
+	defer mockAuthServer.Shutdown()
+	mockAuthServer.options.issuer = mockAuthServer.URL + "/" + defaultProvider
+	refreshProviderConfig(testProviders, mockAuthServer.URL)
+
+	opts := auth.OIDCOptions{Providers: testProviders, DefaultProvider: &defaultProvider}
+	restTesterConfig := RestTesterConfig{SyncFn: channels.DocChannelsSyncFunction, DatabaseConfig: &DatabaseConfig{DbConfig: DbConfig{OIDCConfig: &opts}}}
+
+	// JWT claim based grants do not support named collections
+	restTester := NewRestTesterDefaultCollection(t, &restTesterConfig)
+	require.NoError(t, restTester.SetAdminParty(false))
+	defer restTester.Close()
```

Comment:
I would swap the order of these tests, but I think the default config for `RestTester` is `GuestEnabled: false` and this isn't needed at all.

Reply:
Going to leave this as-is for this PR as it follows pattern used by other OIDC tests, agree it could be updated across the board.